cvat/apps/dextr_segmentation/dextr.py

```diff
@@ -0,0 +1,112 @@
+
+# Copyright (C) 2018 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+from openvino.inference_engine import IENetwork, IEPlugin
+
+import os
+import cv2
+import PIL
+import numpy as np
+
+_IE_CPU_EXTENSION = os.getenv("IE_CPU_EXTENSION", "libcpu_extension_avx2.so")
+_IE_PLUGINS_PATH = os.getenv("IE_PLUGINS_PATH", None)
+
+_DEXTR_MODEL_DIR = os.getenv("DEXTR_MODEL_DIR", None)
+_DEXTR_PADDING = 50
+_DEXTR_TRESHOLD = 0.9
+_DEXTR_SIZE = 512
+
+class DEXTR_HANDLER:
+    _plugin = None
+    _network = None
+    _exec_network = None
+    _input_blob = None
+    _output_blob = None
+    _xml = None
+    _bin = None
+
+    def __init__(self):
+        if not _IE_PLUGINS_PATH:
+            raise Exception("IE_PLUGINS_PATH is not defined")
+        if not _DEXTR_MODEL_DIR:
+            raise Exception("DEXTR_MODEL_DIR is not defined")
+        self._xml = os.path.join(_DEXTR_MODEL_DIR, 'dextr.xml')
+        self._bin = os.path.join(_DEXTR_MODEL_DIR, 'dextr.bin')
+
+    def handle(self, im_path, points):
+        # Lazy initialization
+        if not self._plugin:
+            self._plugin = IEPlugin(device="CPU", plugin_dirs=[_IE_PLUGINS_PATH])
+            self._plugin.add_cpu_extension(os.path.join(_IE_PLUGINS_PATH, _IE_CPU_EXTENSION))
+            self._network = IENetwork.from_ir(model=self._xml, weights=self._bin)
+            self._input_blob = next(iter(self._network.inputs))
+            self._output_blob = next(iter(self._network.outputs))
+            self._exec_network = self._plugin.load(network=self._network)
+
+        image = PIL.Image.open(im_path)
+        numpy_image = np.array(PIL.Image.open(im_path))
```

Comment:
Why are we reading the image again? We can copy `image` or use it directly if it isn't being used anywhere.

Reply:
Yes, I agree.